### ISSUE TITLE
Firebird support

### ISSUE THREAD

**Issue by jcaillon** [liquibase/liquibase#1303]
Hello, Thank you for this great tool.

## Environment
**Liquibase Version**: 3.10.2

**Liquibase Integration & Version**: none, using CLI

**Liquibase Extension(s) & Version**: none

**Database Vendor & Version**: Firebird 3.0.6

**Operating System Type & Version**: Windows 10

## Description
I encounter an issue when using liquibase for a firebird databae.

I'm trying to update my database (which is empty) with a simple changelog and I get the following error:

```log
Unexpected error running Liquibase: conversion error from string "0" [SQLState:22018, ISC error code:335544334] [Failed SQL: (335544334) INSERT INTO DATABASECHANGELOGLOCK (ID, LOCKED) VALUES (1, 0)]
```

Full log:

```log
"C:\Program Files\Java\jdk-11.0.6\bin\java.exe" liquibase.integration.commandline.Main --driver=org.firebirdsql.jdbc.FBDriver --url=jdbc:firebirdsql://localhost:3050/mydb?encoding=NONE --username=sysdba --password=sysdba --contexts OnInitialize --changeLogFile=changelog.xml update
[00:03.717] Liquibase Community 3.10.2 by Datical
[00:03.724] ####################################################
[00:03.727] ##   _     _             _ _                      ##
[00:03.731] ##  | |   (_)           (_) |                     ##
[00:03.733] ##  | |    _  __ _ _   _ _| |__   __ _ ___  ___   ##
[00:03.742] ##  | |   | |/ _` | | | | | '_ \ / _` / __|/ _ \  ##
[00:03.745] ##  | |___| | (_| | |_| | | |_) | (_| \__ \  __/  ##
[00:03.755] ##  \_____/_|\__, |\__,_|_|_.__/ \__,_|___/\___|  ##
[00:03.759] ##              | |                               ##
[00:03.763] ##              |_|                               ##
[00:03.767] ##                                                ## 
[00:03.771] ##  Get documentation at docs.liquibase.com       ##
[00:03.776] ##  Get certified courses at learn.liquibase.com  ## 
[00:03.781] ##  Get advanced features and support at          ##
[00:03.785] ##      liquibase.com/protrial                    ##
[00:03.790] ##                                                ##
[00:03.795] ####################################################
[00:03.799] Starting Liquibase at Thu, 06 Aug 2020 10:59:49 CEST (version 3.10.2 #22 built at Mon Jul 27 04:21:02 UTC 2020)
[00:04.901] Unexpected error running Liquibase: conversion error from string "0" [SQLState:22018, ISC error code:335544334] [Failed SQL: (335544334) INSERT INTO DATABASECHANGELOGLOCK (ID, LOCKED) VALUES (1, 0)]
```

## Steps To Reproduce
Try to update the firebird database with the following changelog file (this is an example, it will not work for any changelog):

```xml
<?xml version="1.1" encoding="UTF-8" standalone="no"?>
<databaseChangeLog logicalFilePath="yadaman" xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:pro="http://www.liquibase.org/xml/ns/pro" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/pro http://www.liquibase.org/xml/ns/pro/liquibase-pro-3.9.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.9.xsd">
    <changeSet author="jcaillon (generated)" id="1596640136884-1">
        <createTable tableName="DEPARTMENT">
            <column name="DEPTNO" type="BIGINT">
                <constraints nullable="false" primaryKey="true" primaryKeyName="DEPT_PK"/>
            </column>
            <column name="NAME" type="VARCHAR(14)"/>
            <column name="LOCATION" type="VARCHAR(13)"/>
        </createTable>
    </changeSet>
</databaseChangeLog>
```

## Work around
A workaround is to manually create the DATABASECHANGELOGLOCK table using SMALLINT type instead of BOOLEAN:

```sql
CREATE TABLE DATABASECHANGELOGLOCK (ID VARCHAR(255) NOT NULL, LOCKED SMALLINT NOT NULL, LOCKGRANTED TIMESTAMP, LOCKEDBY VARCHAR(255));
```

Thus the bug is simply either that the schema should be created in smallint or that the update sql generated should be true/false instead of 0/1.

**Comment by molivasdat** [comment]
Hi @jcaillon Thanks for the detailed issue description.  Would it be possible to create a fix in a PR of the code?  If not, we can add it to the list of issues that we are processing.

**Comment by nvoxland** [comment]
This appears to be a duplicate of #1872 which is being fixed in #1875
